code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -168,6 +169,24 @@
 				if(faction_check && !attack_same)
 					return 0
 			return 1
+
+		if(istype(the_target, /obj/mecha))
+			var/obj/mecha/M = the_target
+			if(M.occupant)//Just so we don't attack empty mechs
+				if(CanAttack(M.occupant))
+					return 1
+
+		if(istype(the_target, /obj/machinery/porta_turret))
+			var/obj/machinery/porta_turret/P = the_target
+			if(P.faction in faction)
+				return 0
+			if(P.has_cover &&!P.raised) //Don't attack invincible turrets
```

Comment:
He does not actually mean "invincible" in the same way you do, Kor's dictionary is slightly different in some cases. "Inactive" is a better term in this instance.

 Here, the turret would be neutral to the mob, as it is not configured to fire on it, or is unable to do so for some reason (such as being powered off). 

An example would be if a Xenobiologist filled the AI's chamber with gold slime core mobs, they would not waste time attacking the turrets unless they were an actual threat. This is not the best example, but it is meant to cover a lot of things.